[nramapurath] iP

nramapurath opened the pull request:
# DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
* easy to learn
+ ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

--- 

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

nramapurath committed:
Implement basic echo functionality

nramapurath committed:
Implement adding and listing tasks

nramapurath committed:
Add javadocs in Duke.java

nramapurath committed:
Add class to represent tasks

nramapurath committed:
Implement marking tasks

nramapurath committed:
Add support for task types

nramapurath committed:
Add basic tests for Duke

nramapurath committed:
Modify print indentations

nramapurath committed:
Refactor task creation into separate functions

nramapurath committed:
Create DukeException

nramapurath committed:
Error handling for empty todo and invalid input

nramapurath committed:
Correct test case to match new error message

nramapurath committed:
Add support for deleting tasks

nramapurath committed:
Add test for deleting tasks

nramapurath committed:
Fix indentation in switch statement

nramapurath committed:
Add save functionality

nramapurath committed:
Add date time support

nramapurath committed:
Merge branch 'branch-Level-7'

nramapurath committed:
Merge branch 'branch-Level-8'

nramapurath committed:
Fix datetime storage format

nramapurath committed:
Refactor most interactions to Ui class

nramapurath committed:
Refactor user command parsing to Parser class

nramapurath committed:
Edit test files

nramapurath committed:
Divide classes into packages

nramapurath committed:
Merge remote-tracking branch 'origin/add-gradle-support'

nramapurath committed:
Change application main class name to match directory structure

nramapurath committed:
Add JUnit Tests for Tasks and TaskList

nramapurath committed:
Add JavaDoc comments

nramapurath committed:
Edit to comply with coding standard

nramapurath committed:
Implement find task functionality

nramapurath committed:
Merge JavaDoc branch

nramapurath committed:
Fix merge conflict between JavaDocs and Coding Standard branch

nramapurath committed:
Fix merge conflict with Level-9 branch

potty10 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,105 @@
+                    ui.listTasks(tasks);
+                } else {
+                    try {
+                        switch (parser.getCommand()) {
```
I'm not sure if having indentations in this try block is a coding standard violation.

potty10 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,105 @@
+    }
+
+    /**
+     * Performs ui operations according to parsed user inputs.
+     * If user inputs "bye", store list of tasks and exit.
+     * If user inputs "list", print current tasks.
+     * If user inputs "mark NUMBER" or "unmark NUMBER", update the status of that task number.
+     * If user inputs a task, add to current tasks.
```
I like how you made it simple to understand this method. Nice work!

potty10 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,210 @@
+    /**
+     * Prints Duke's farewell message (bounded by lines above and below).
+     */
+    public void farewell() {
```
I think it will be hard to deduce that this method prints a farewell message. Maybe a better variable name is `printFarewellMessage`?

potty10 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,47 @@
+ */
+public class Deadline extends Task {
+
+    protected LocalDateTime by;
```
Maybe adding a Javadoc here is helpful? Or do you think renaming the variable is a better choice?

potty10 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,210 @@
+    }
+
+    /**
+     * Lists tasks matching keyword.
```
Perhaps a better method description is "Lists tasks whose names contains the keyword"?

potty10 reviewed (commented):
Overall, it is well-written code that follows the OOP principles. Keep up the good work!

nramapurath committed:
Use CheckStyle

nramapurath committed:
Add JavaFX and GUI (in sample echo code)

nramapurath committed:
Implement Duke through GUI

nramapurath committed:
Make minor visual tweaks

nramapurath committed:
Merge branch 'branch-Level-10'
Merges branch with GUI functionality.

nramapurath committed:
Add assertions

nramapurath committed:
Improve code quality

nramapurath committed:
Merge pull request #3 from nramapurath/branch-A-Assertions

Add assertions

nramapurath committed:
Merge branch 'master' into branch-A-CodeQuality
Merge updated upstream (updates from Assertions branch)

branch-A-Assertions has been merged into the remote master branch.
So, to sync changes, let's merge the synced master tto the CodeQuality
branch before proceeding with the next PR.

nramapurath committed:
Merge pull request #4 from nramapurath/branch-A-CodeQuality

Improve code quality

nramapurath committed:
Edit build.gradle to fix jar

nramapurath committed:
Implement tagging tasks

Currently, there is no way for users to tag their tasks under
specific categories.

We shall add the functionality to tag tasks with a tag of
user's choosing (any string input).

Let's
* update the task storage format to accommodate tag storage
* update task constructors to store tags
* update task print methods to display tags (if any)
* add methods to parse tags from user input

In the future, we can look into restricting the length of tags.

nramapurath committed:
Merge branch 'branch-C-Tagging'

Merges branch with tagging functionality.

nramapurath committed:
Add more error handling

Currently, there is a lack of error handling, such as when user
enters date in an incorrect format.

To fix this, let's:
* have constructors throw DukeExceptions with the appropriate
error messages.
* display informative text to the user on reaching such states
to correct their next input.

nramapurath committed:
Merge branch 'branch-A-MoreErrorHandling'

nramapurath committed:
Add screenshot to docs

nramapurath committed:
Update user guide

nramapurath committed:
Update UG for exit instructions

nramapurath committed:
Remove unused statements

nramapurath committed:
Fix JavaFX truncating Label text
